cmake/modules/RootConfiguration.cmake

```diff
@@ -495,6 +495,11 @@ if(cxx17)
 else()
   set(usec++17 undef)
 endif()
+if(runtime_cxxmodules)
+  set(usecxxmodules define)
```

Comment:
I'm a bit concerned that we entangle the idea of "use and build C++ modules when building ROOT, using the compiler you want to build ROOT with" and "use C++ modules for dictionaries, using the clang libraries from interpreter/llvm".

Can you clarify that maybe by more explicit names, e.g. here `usedictcxxmodues`?

Reply:
Well, it's not just dictionaries that have modules, but also all external libraries, so that's rather confusing with a dict in the name.

Maybe we just take over the original flag we have (`cxxmodules`) because that's the most obvious thing for the central idea: Have C++ modules in ROOT. And we always called the project "C++ modules in ROOT", so that name fits to how we presented it to the outside.

The compile time modules which is only interested for people using the most recent clang can be `compile_with_modules` or something like that.

Reply:
Axel, the variable usecxxmodules is temporary. Once everything works (which IMO is not far from now) we will remove it and make it default behavior.

I do not think we should spend time in making the swap of the compiletime/runtime cxxmodules.